# Information about installed software

The following software is currently installed (please follwo the links to learn more):

  * [ARAGORN](#aragorn)
  * [Binfo seminar toolbox](#binfo-seminar-toolbox)
  * [Barrnap](#barrnap)
  * [HMMER](#hmmer)
  * [Infernal](#infernal)
  * [Kaiju](#kaiju)
  * [Parsnp](#parsnp)
  * [Prodigal](#prodigal)
  * [Prokka](#prokka)
  * [QUAST](#quast)
  * [SPAdes](#spades)
  * [SRA Toolkit](#sra-toolkit)
  * [Tbl2asn](#tbl2asn)

## ARAGORN

### Purpose

ARAGORN identifies tRNA and tmRNA genes and predicts tRNA secondary structure. 

### Usage

Type __aragorn -h__ to learn about how to use the program.

### Download

Download ARAGORN from the homepage of the [University of Lund](http://mbio-serv2.mbioekol.lu.se/ARAGORN/Downloads/).

### Miscellaneous

An introductory paper has been published in [Nucleic Acids Research](https://academic.oup.com/nar/article/32/1/11/1194008/ARAGORN-a-program-to-detect-tRNA-genes-and-tmRNA).

## Binfo seminar toolbox

A collection of shell scripts provided by Matthias Steglich with DSMZ available at https://github.com/dghm-genomics/Binfo_seminar_toolbox.

### Purpose

A collection of convenience shell scripts. 

### Usage

Type __bwamapping.sh__, __fas-cut.sh__, __fas-get-variable-sites.sh__, __fas-info.sh__, __fas-selector.sh__, __fastq-coverage.sh__, or __vs-varcall.sh__ to learn about the usage of each of the scripts.

### Download

Download Binfo seminar toolbox from [Github](https://github.com/dghm-genomics/Binfo_seminar_toolbox).

### Miscellaneous

The scripts were created by Matthias Steglich (DSMZ).

## BBTools

### Purpose

BBTools is a free and open-source suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data.

### Usage

Type __bbduk.sh__, __bbmap.sh__, __bbmerge.sh__, or __reformat.sh__ to learn about the usage of each of the programs within the BBTools suite.

### Download

BBTools can be downloaded from [Sourceforge](https://sourceforge.net/projects/bbmap/)

### Miscellaneous

BB stands for Bestus Bioinformaticus. BBTools is hosted at the [Joint Genome Institute](http://jgi.doe.gov/data-and-tools/bbtools/).

## Barrnap

### Purpose

Barrnap predicts the location of ribosomal RNA genes in genomes of bacteria, archaea, mitochondria, and eukaryotes. 

### Usage

Type __barrnap__ to learn about the program's usage.

### Download

Barrnap can be downloaded from [GitHub](https://github.com/tseemann/barrnap).

### Miscellaneous

Barrnap was created by Torsten Seemann, who has provided a [manual](https://github.com/tseemann/barrnap/blob/master/README.md).

## HMMER

### Purpose

HMMER is a collection of programs used for searching sequence databases for sequence homologs, and for making sequence alignments. 

### Usage

Type __hmmbuild__, __hmmalign__, __phmmer__, __jackhmmer__, __hmmsearch__, __hmmscan__, __hmmpgmd__, __nhmmer__, __nhmmscan__, __alimask__, __hmmconvert__, __hmmemit__, __hmmfetch__, __hmmpress__, or __hmmstat__ to learn about the programs in HMMER.

### Download

HMMER can be downloaded from [Eddylab](http://eddylab.org/software/hmmer3/CURRENT/).

### Miscellaneous

Eddylab is located at the Department of Molecular & Cellular Biology, Harvard University, Cambridge, MA.

## Infernal

### Purpose

Infernal ("INFERence of RNA ALignment") is a collection of programs for searching DNA sequence databases for RNA structure and sequence similarities.

### Usage

Type __cmalign__, __cmbuild__, __cmcalibrate__, __cmvonvert__, __cmemit__, __cmfetch__, __cmpress__, __cmscan__, __cmsearch__, or __cmstat__ to learn about the programs in Infernal.

### Download

Infernal can be downloaded from [Eddylab](http://eddylab.org/infernal/).

### Miscellaneous

Visit the [Infernal Homepage](http://eddylab.org/infernal/) for further information. 

## Kaiju

### Purpose

Kaiju is a program for sensitive taxonomic classification of high-throughput sequencing reads from metagenomic whole genome sequencing experiments. 

### Usage

Type __kaiju__ to learn about its usage.

### Download

Kaiju can be downloaded from [GitHub](https://github.com/bioinformatics-centre/kaiju).

### Miscellaneous

Please refer to the [Bioinformatics Centre of the University of Copenhagen](http://kaiju.binf.ku.dk) for further information.

## Parsnp 

### Purpose

Parsnp is a command-line-tool for efficient microbial core genome alignment and SNP detection. 

### Usage

Type __parsnp__ to learn about its usage.

### Download

Parsnp can be downloaded from [GitHub](https://github.com/marbl/parsnp).

### Miscellaneous

Documentation is available at [Readthedocs](http://harvest.readthedocs.io/en/latest/).

## Prodigal

### Purpose

Prodigal (Prokaryotic Dynamic Programming Genefinding Algorithm) is a microbial (bacterial and archaeal) gene finding program.

### Usage

Type __prodigal__ to learn about its usage.

### Download

Prodigal can be downloaded from [GitHub](https://github.com/hyattpd/prodigal).

### Miscellaneous

Prodigal was developed at Oak Ridge National Laboratory and the University of Tennessee. Please visit the [wiki page](https://github.com/hyattpd/prodigal/wiki) for further information.

## Prokka 

### Purpose

Prokka is a software tool for the rapid annotation of prokaryotic genomes.

### Usage

Type __prokka__ to learn about its usage.

### Download

Prokka can be downloaded from [GitHub](https://github.com/tseemann/prokka).

### Miscellaneous

Prokka was created by Torsten Seemann.

## QUAST 

### Purpose

QUAST performs fast and convenient quality evaluation and comparison of genome assemblies. 

### Usage

Type __quast.py__ to learn about its usage.

### Download

QUAST can be downloaded from [Sourceforge](http://quast.sourceforge.net/download.html).

### Miscellaneous

Please visit the [QUAST page at Sourceforge](http://quast.sourceforge.net) for further information. A [web-accessible version](http://quast.bioinf.spbau.ru) of QUAST is also available.

## SPAdes

### Purpose

SPAdes is a genome assembler that has gained popularity for the assembly of prokaryotic genomes.

### Usage

Type __spades.py__ to learn about its usage.

### Download

SPAdes can be downloaded from the [Center for Algorithmic Biotechnology](http://cab.spbu.ru/software/spades) at St. Petersburg State University.

### Miscellaneous

Please visit [the SPAdes website](http://cab.spbu.ru/software/spades) for further information.

## SRA Toolkit

### Purpose

The [Sequence Read Archive (SRA)](https://trace.ncbi.nlm.nih.gov/Traces/sra/) stores raw sequence data from "next-generation" sequencing technologies. The [SRA toolkit](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software) allows download and conversion of raw data into different formats.

### Usage

Type __fastq-dump__, __prefetch__, __sam-dump__, __sra-pileup__, __vdb-config__, and __vdb-decrypt__ to learn about the use of the programs within the toolkit.

### Download

The SRA Toolkit can be downloaded from [NCBI](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software).

### Miscellaneous

Accession no. [ERR486835](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=ERR486835) and [ERR486836](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=ERR486836) represent two randomly chosen Illumina MiSeq paired end sequencing reads of *Mycoplasma genitalium*. They have been downloaded to directory /home/workshop-user/ncbi/public/sra on the image.

## Tbl2asn

### Purpose

Tbl2asn is a command-line program that automates the creation of sequence records for submission to GenBank.

### Usage

Type __tbl2asn --help__ to learn about the program's usage.

### Download

Tbl2asn can be downloaded from [NCBI](https://www.ncbi.nlm.nih.gov/genbank/tbl2asn2/).

### Miscellaneous

Further information is available at [NCBI](https://www.ncbi.nlm.nih.gov/genbank/tbl2asn2/).

